src/caret_analyze/architecture/architecture.py

```diff
@@ -196,23 +196,24 @@ def _verify(nodes: Collection[NodeStruct]) -> None:
                      f'callback_type: {uniqueness_violated[0]}'
                      f'period_ns: {uniqueness_violated[1]}'))
 
-
-"""
-    def rename_callback(src: str, dest: str):
+    def rename_callback(self, src: str, dst: str):
         raise NotImplementedError('')
 
-    def rename_node(src: str, dest: str):
-        raise NotImplementedError('')
+    def rename_node(self, src: str, dst: str):
+        n: NodeStruct = Util.find_one(lambda x: x.node_name == src, self._nodes)
```

Comment:
Util.find_one throws an exception when no target is found instead of returning None.
see: https://github.com/tier4/CARET_analyze/blob/4c72d9756dd3fa86f27958b821c5a0eb90cf6ef4/src/caret_analyze/common/util.py#L57

And It's better to use Util.find_similar_one to support notification for similar node name.
https://github.com/tier4/CARET_analyze/blob/main/src/caret_analyze/common/util.py#L91